actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -128,18 +128,21 @@ def stylesheet_link_tag(*sources)
       #   # => <link rel="alternate" type="application/rss+xml" title="RSS" href="http://www.currenthost.com/news/feed" />
       #   auto_discovery_link_tag(:rss, "http://www.example.com/feed.rss", {title: "Example RSS"})
       #   # => <link rel="alternate" type="application/rss+xml" title="Example RSS" href="http://www.example.com/feed" />
+      #   auto_discovery_link_tag(:atom, {controller: "news", action: "feed"}, {data: {'turbolinks-track' => true}})
+      #   # => <link rel="alternate" type="application/atom+xml" title="ATOM" href="http://www.currenthost.com/news/feed" data-turbolinks-track="true" />
       def auto_discovery_link_tag(type = :rss, url_options = {}, tag_options = {})
         if !(type == :rss || type == :atom) && tag_options[:type].blank?
           raise ArgumentError.new("You should pass :type tag_option key explicitly, because you have passed #{type} type other than :rss or :atom.")
         end
 
-        tag(
-          "link",
+        tag_options.merge!({
```

Comment:
we better not modify the passed in options hash. This could cause unintended side-effects. What about modifying the hash containing the default options?

```ruby
{}.reverse_merge!(tag_options)
```